### PR TITLE
Migrate to Micronaut 4.0.0 and build plugins to 6.2.0

### PR DESCRIPTION
**Pull request by wetted**

### CODE REVIEW

File: buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-api.gradle [hunk 1]
```diff
@@ -3,16 +3,10 @@ plugins {
 }
 
 dependencies {
-    annotationProcessor(mn.micronaut.serde.processor)
-    api(mn.micronaut.serde.api)
-    api(libs.jackson.annotations)
+    annotationProcessor(mnSerde.micronaut.serde.processor)
+    api(mnSerde.micronaut.serde.api)
+    api(mn.jackson.annotations)
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-api.gradle:8]:
```suggestion
    api(mnSerde.micronaut.serde.jackson)
```

File: buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-api.gradle [hunk 2]
```diff
@@ -3,16 +3,10 @@ plugins {
...
     implementation(mn.micronaut.validation)
-}
 
-configurations.all {
-    resolutionStrategy.dependencySubstitution {
-        substitute(module("io.micronaut:micronaut-jackson-databind"))
-                .using(module("io.micronaut.serde:micronaut-serde-jackson:${mn.versions.micronaut.serialization.get()}"))
-    }
+    testImplementation(mnSerde.micronaut.serde.api)
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-api.gradle:11]: We shouldn't need this as it's an `api` dependency already

File: buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-base.gradle [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    mavenCentral()
+    maven { url "https://s01.oss.sonatype.org/content/repositories/snapshots/" }
+}
+
+configurations.all {
+    resolutionStrategy.dependencySubstitution {
+        substitute(module("io.micronaut:micronaut-jackson-databind"))
+                .using(module("io.micronaut.serde:micronaut-serde-jackson:${libs.versions.micronaut.serde.get()}"))
+    }
+}
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-base.gradle:11]: Do we need this?

File: buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-module.gradle [hunk 4]
```diff
@@ -1,14 +1,16 @@
 plugins {
+    id "io.micronaut.build.internal.chatbots-base"
     id "io.micronaut.build.internal.module"
 }
+
 dependencies {
     implementation(mn.micronaut.validation)
+
+    testImplementation(mn.micronaut.jackson.databind)
```
Comment (timyates) [inline buildSrc/src/main/groovy/io.micronaut.build.internal.chatbots-module.gradle:9]:
```suggestion
    testImplementation(mnSerde.micronaut.serde.jackson)
```

**Review by timyates — CHANGES_REQUESTED** [review]

**Review by timyates — APPROVED** [review]

**Review by sdelamo — APPROVED** [review]

**Pull request merged**